bool[8] zones
float32 angle